src/components/spreadsheet/utils/equipment-table-editors.js

```diff
@@ -104,6 +105,99 @@ export const GeneratorRegulatingTerminalEditor = forwardRef(
+                    getValue: () => {
+                        return getTapChangerRegulationTerminalValue(
+                            gridContext.dynamicValidation?.[tapChangerType] ||
+                                {}
+                        );
+                    },
+                    getField: () => {
+                        return colDef.field;
+                    },
+                };
+            },
+            [colDef.field, gridContext.dynamicValidation, tapChangerType]
+        );
+
+        const handleSaveRegulatingTerminalPopup = (
+            updatedRegulatedTerminal
+        ) => {
+            const {
+                equipment: { type: equipmentType, id: equipmentId } = {},
+                voltageLevel: { id: voltageLevelId } = {},
+            } = updatedRegulatedTerminal || {};
+            gridContext.dynamicValidation = deepUpdateValue(
+                gridContext.dynamicValidation,
+                `${tapChangerType}.regulatingTerminalConnectableId`,
+                equipmentId
+            );
+            gridContext.dynamicValidation = deepUpdateValue(
+                gridContext.dynamicValidation,
+                `${tapChangerType}.regulatingTerminalConnectableType`,
+                equipmentType
+            );
+            gridContext.dynamicValidation = deepUpdateValue(
+                gridContext.dynamicValidation,
+                `${tapChangerType}.regulatingTerminalVlId`,
+                voltageLevelId
+            );
+            setOpenTWTRegulatingTerminalPopup(false);
+        };
+        const handleCancelRegulatingTerminalPopup = () => {
+            setOpenTWTRegulatingTerminalPopup(false);
+            gridApi.stopEditing();
+        };
+
+        const getTapChangerValue = () => {
+            const tapChanger = {
+                ...rowData?.[tapChangerType],
+                regulatingTerminalConnectableId:
+                    rowData?.[tapChangerType]
+                        ?.regulatingTerminalConnectableId || '',
+                regulatingTerminalConnectableType:
+                    rowData?.[tapChangerType]
+                        ?.regulatingTerminalConnectableType || '',
+                regulatingTerminalVlId:
+                    rowData?.[tapChangerType]?.regulatingTerminalVlId || '',
+            };
+            return tapChanger;
+        };
+
+        return (
+            <RegulatingTerminalModificationDialog
+                open={openTWTRegulatingTerminalPopup}
+                onClose={handleCancelRegulatingTerminalPopup}
+                currentNode={gridContext.currentNode}
+                studyUuid={gridContext.studyUuid}
+                onModifyRegulatingTerminalGenerator={(
+                    updatedRegulatedTerminal
+                ) => {
+                    handleSaveRegulatingTerminalPopup(updatedRegulatedTerminal);
+                }}
+                data={getTapChangerValue()}
+                previousData={gridContext.dataToModify}
```

Comment:
The previous values are not displayed because they are not mapped the same way as the generators